packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -312,8 +312,8 @@ class ThemeData with Diagnosticable {
     secondaryHeaderColor ??= isDark ? Colors.grey[700] : primarySwatch[50];
     textSelectionColor ??= isDark ? accentColor : primarySwatch[200];
     // TODO(hansmuller): We need a TextFieldTheme to handle this instead, https://github.com/flutter/flutter/issues/56082
-    cursorColor = cursorColor ?? const Color.fromRGBO(66, 133, 244, 1.0);
-    textSelectionHandleColor ??= isDark ? Colors.tealAccent[400] : primarySwatch[300];
+    cursorColor ??= isDark ? Colors.tealAccent[200] : primarySwatch;
+    textSelectionHandleColor ??= isDark ? Colors.tealAccent[200] : primarySwatch;
```

Comment:
Can you provide a reference for the color choices you have made throughout this change? Did they come from the material spec?

Reply:
The color choices generally come from platform-specific material specs.

For the checkColor, in making it themable, I chose canvasColor as the default check color is the same as "?attr/colorSurface" according to what I understood from this page: https://developer.android.com/reference/com/google/android/material/checkbox/MaterialCheckBox and by looking at the many examples in the design spec (https://material.io/components/selection-controls). In fact, there doesn't seem to be a provision for theming this separately on android and the control might benefit from animating the check as a cutout rather than an overlay. However, this would be a more comprehensive change in connection with the Material Theme overhall being worked on, so I made this small change as a stopgap.

As for the text cursor, this should default to primary according to both the android and iOS material specs for text inputs (Links from here: https://material.io/components/text-fields). While there is no specific documentation on the selection handle color, this always matches the collapsed cursor color in tests. This is especially obvious on iOS where the handle extends upwards to bookend selections. Both according to the spec and as confirmed on #56082, this is designed to match the colorControlActivated color used for focused text field borders. Again, I created this pull mainly as a stopgap.

As for the dark theme colors, I updated everything to match the accentColor (teal[200]) for consistency since this is the current default for the textSelectionHandleColor. These should arguably be the same as primary, however the current dark theme primary is not a color despite Material dark theme guidelines (https://material.io/design/color/dark-theme.html#anatomy) suggesting it should be (a desaturated version of light primary). Changing the entire default dark theme to better match the guidelines was out of scope for this pull which is just about clearing out unthemable colors.